packages/scenes/src/components/SceneTimeRangeCompare.tsx

```diff
@@ -117,9 +117,15 @@ export class SceneTimeRangeCompare
     return extraQueries;
   }
 
-  // The query runner should rerun the comparison query if the compareWith value has changed.
-  public shouldRerun(prev: SceneTimeRangeCompareState, next: SceneTimeRangeCompareState): boolean {
-    return prev.compareWith !== next.compareWith;
+  // The query runner should rerun the comparison query if the compareWith value has changed and there are queries that haven't opted out of TWC
+  public shouldRerun(
+    prev: SceneTimeRangeCompareState,
+    next: SceneTimeRangeCompareState,
+    queries: SceneDataQuery[]
+  ): boolean {
+    return (
+      prev.compareWith !== next.compareWith && queries.filter((query) => query.timeRangeCompare !== false).length > 0
```

Comment:
tiny nit: can use find and exit early if we find any

```suggestion
      prev.compareWith !== next.compareWith && queries.find((query) => query.timeRangeCompare !== false) !== undefined
```